File: pythons/nomor.py
valid = False

while not valid:
    num = input("Masukkan nomor:")
    
    # CEK SPASI
    if " " in num:
        print("Error: Nomor telepon tidak boleh mengandung spasi")
        continue
    
    # CEK FORMAT
    if num.startswith("+62"):
        fltdnum=num[3:]
    elif num.startswith("62"):
        fltdnum=num[2:]
    elif num.startswith("0"):
        fltdnum=num[1:]
    else:
        print("Error: format tidak valid")
        continue
    
    # CEK ANGKA
    if not fltdnum.isdigit():
        print("Error: Input wajib berupa angka")
        continue
    
    # CEK PANJANG
    if len(fltdnum) < 10 or len(fltdnum) > 13:
        print("Error: Panjang digit harus di range 10-13")
        continue
    
    # CEK DUA DIGIT AWAL
    dua_digit=int(fltdnum [:2])
    
    if dua_digit < 11 or dua_digit > 99:
        print(f"Error: Dua digit awal ({fltdnum[:2]}) tidak valid")
        continue
    
    # VALIDASI AKHIR
    valid = True
    print(f"Valid! Nomor {num} memenuhi semua syarat.")

File: javas/Nomor.java
import java.util.Scanner;

public class Nomor {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        boolean valid = false;

        while (!valid) {
            System.out.print("Masukkan nomor:");
            String num = sc.nextLine();

            if (num.contains(" ")) {
                System.out.println("Error: Nomor telepon tidak boleh mengandung spasi");
                continue;
            }

            String fltdNum;
            if (num.startsWith("+62")) {
                fltdNum = num.substring(3);
            } else if (num.startsWith("62")) {
                fltdNum = num.substring(2);
            } else if (num.startsWith("0")) {
                fltdNum = num.substring(1);
            } else {
                System.out.println("Error: format tidak valid");
                continue;
            }

            if (!fltdNum.matches("\\d+")) {
                System.out.println("Error: Input wajib berupa angka");
                continue;
            }

            if (fltdNum.length() < 10 || fltdNum.length() > 13) {
                System.out.println("Error: Panjang digit harus di range 10-13");
                continue;
            }

            int duaDigit = Integer.parseInt(fltdNum.substring(0, 2));

            if (duaDigit < 11 || duaDigit > 99) {
                System.out.println("Error: Dua digit awal (" + fltdNum.substring(0, 2) + ") tidak valid");
                continue;
            }

            valid = true;
            System.out.println("Valid! Nomor " + num + " memenuhi semua syarat.");
        }

        sc.close();
    }
}
